Clue.js

```diff
@@ -0,0 +1,122 @@
+  color: "blue",
+  description: "She is from a wealthy family and uses her status and money to earn popularity",
+  age: "36",
+  image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+  occupation: "Socialité"
+};
+var mrMustard = {
+  first_name: "Jack",
+  last_name: "Mustard",
+  color: "yellow",
+  description: "He is a former football player who tries to get by on his former glory",
+  age: "62",
+  image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+  occupation: "Retired Football player"
+};
+//Characters' Collection
+var charactersArray = [];
+charactersArray.push(mrGreen, drOrchid, profPlum, missScarlet, mrsPeacock, mrMustard);
+// Weapons
+var rope        = { name: "rope", weight: 10 };
+var knife       = { name: "knife", weight: 8 };
+var candlestick = { name: "candlestick", weight: 2 };
+var dumbbell    = { name: "dumbbell", weight: 30 };
+var poison      = { name: "poison", weight: 2 };
+var axe         = { name: "axe", weight: 15 };
+var bat         = { name: "bat", weight: 13 };
+var trophy      = { name: "trophy", weight: 25 };
+var pistol      = { name: "pistol", weight: 20 };
+//Weapons' Collection
+var weaponsArray = [];
+weaponsArray.push(rope, knife, candlestick, dumbbell, poison, axe, bat, trophy, pistol);
+//Rooms
+var diningRoom   = { name: "Dinning Room" };
+var conservatory = { name: "Conservatory" };
+var kitchen      = { name: "Kitchen" };
+var study        = { name: "Study" };
+var library      = { name: "Library" };
+var billiardRoom = { name: "Billiard Room" };
+var lounge       = { name: "Lounge" };
+var ballroom     = { name: "Ballroom" };
+var hall         = { name: "Hall" };
+var spa          = { name: "Spa" };
+var livingRoom   = { name: "Living Room" };
+var observatory  = { name: "Observatory" };
+var theater      = { name: "Theater" };
+var  guestHouse  = { name: "Guest House" };
+var patio        = { name: "Patio" };
+//Rooms' Collection
+var roomsArray = [];
+roomsArray.push(diningRoom, conservatory, kitchen, study, library, billiardRoom, lounge, ballroom, hall, spa, livingRoom, observatory, theater, guestHouse, patio);
+//Iteration 1. Random mistery
+function random_selector(array) {
+  var result;
+  do result = Math.floor(Math.random()*array.length)
+  while(result === array.legth);
+  return array[result];
+}
+
+function callRandomSelection(){
+  var selection=[];
+  selection.push(random_selector(charactersArray));
+  selection.push(random_selector(roomsArray));
+  selection.push(random_selector(weaponsArray));
+  return selection;
+}
+
+function pick_mistery() {
+  var selection = callRandomSelection();
+  var envelope = {
+    suspect: selection[0],
+    place: selection[1],
+    weapon: selection[2]
+  }
+  return envelope;
+}
+
+function reveal_mistery(mistery) {
+  console.log("The guilty character is " + mistery.suspect +
+              ", who committed the murder in the " + mistery.place + " with the weapon " + mistery.weapon);
```

Comment:
Estáis llamando a cada objeto, pero no aparece nada.
Por ejemplo, en "mistery.suspect" , **mistery** es el objeto y **suspect** es una propiedad de **mistery**. **suspect** es un objeto también, por lo que habrá que llamar a su propiedad para mostrar algo. En este caso **name**
**mistery.suspect.name**